discord/types/channel.py

```diff
@@ -153,13 +153,14 @@ class ForumChannel(_BaseTextChannel):
 class DMChannel(_BaseChannel):
     type: Literal[1]
     last_message_id: Optional[Snowflake]
-    recipients: List[PartialUser]
+    recipients: NotRequired[List[PartialUser]]
```

Comment:
This and the change below it seem motivated purely by the interaction channel type changes, instead of changing the overall type to fit this it might be better to tentatively create a new DM channel type for interaction DM channel.